llvm/lib/Target/RISCV/RISCVISelLowering.cpp

```diff
@@ -21951,7 +21951,8 @@ bool RISCVTargetLowering::lowerDeinterleaveIntrinsicToLoad(
         Intrinsic::riscv_vlseg6, Intrinsic::riscv_vlseg7,
         Intrinsic::riscv_vlseg8};
 
-    unsigned SEW = ResVTy->getElementType()->getScalarSizeInBits();
+    unsigned SEW =
+        DI->getDataLayout().getTypeSizeInBits(ResVTy->getElementType());
```

Comment:
Can we save the DataLayout in a variable? We used it earlier for the `isLegalInterleavedAccessType` check.

Reply:
Sure we can~